qemu/qemu.go

```diff
@@ -284,33 +299,51 @@ func (object Object) Valid() bool {
 func (object Object) QemuParams(config *Config) []string {
 	var objectParams []string
 	var deviceParams []string
+	var machineParams []string
 	var qemuParams []string
 
-	deviceParams = append(deviceParams, string(object.Driver))
-	deviceParams = append(deviceParams, fmt.Sprintf(",id=%s", object.DeviceID))
-
 	switch object.Type {
 	case MemoryBackendFile:
 		objectParams = append(objectParams, string(object.Type))
 		objectParams = append(objectParams, fmt.Sprintf(",id=%s", object.ID))
 		objectParams = append(objectParams, fmt.Sprintf(",mem-path=%s", object.MemPath))
 		objectParams = append(objectParams, fmt.Sprintf(",size=%d", object.Size))
 
+		deviceParams = append(deviceParams, string(object.Driver))
+		deviceParams = append(deviceParams, fmt.Sprintf(",id=%s", object.DeviceID))
 		deviceParams = append(deviceParams, fmt.Sprintf(",memdev=%s", object.ID))
 	case TDXGuest:
 		objectParams = append(objectParams, string(object.Type))
 		objectParams = append(objectParams, fmt.Sprintf(",id=%s", object.ID))
 		if object.Debug {
 			objectParams = append(objectParams, ",debug=on")
 		}
+		deviceParams = append(deviceParams, string(object.Driver))
+		deviceParams = append(deviceParams, fmt.Sprintf(",id=%s", object.DeviceID))
 		deviceParams = append(deviceParams, fmt.Sprintf(",file=%s", object.File))
+	case SevGuest:
+		objectParams = append(objectParams, string(object.Type))
+		objectParams = append(objectParams, fmt.Sprintf(",id=%s", object.ID))
+		objectParams = append(objectParams, fmt.Sprintf(",cbitpos=%s", object.CBitPos))
+		objectParams = append(objectParams, fmt.Sprintf(",reduced-phys-bits=%s", object.ReducedPhysBits))
+
+		machineParams = append(machineParams, fmt.Sprintf("memory-encryption=%s", object.ID))
```

Comment:
I think it would be preferable to use the newer `confidential-guest-support=` machine parameter instead of `memory-encryption=`.  The new one will extend better to non-x86 and non-encryption-based schemes for implementing confidential guests.

Reply:
Thanks @dgibson, I'll look into simplifying these changes with `confidential-guest-support`

@sandeepg-gupta had you considered using `confidential-guest-support` in your prototype? My past experience with SEV VMs was limited to the `memory-encryption=` parameter.

Reply:
That's not surprising, `confidential-guest-support` is pretty new.

Reply:
@dgibson I've updated the patch to use `confidential-guest-support`